From 7aa805a0947c7b899edd2db1b6db42f6af9a2664 Mon Sep 17 00:00:00 2001
From: Darek Stojaczyk <dariusz.stojaczyk@intel.com>
Date: Fri, 11 Oct 2019 07:06:35 -0400
Subject: [PATCH 3/8] config: comment out vhost

SPDK can be configured to build without vhost target,
so we shouldn't always need to build DPDK rte_vhost.

Comment out rte_vhost build option, as well as build
options for vhosts's dependencies, so now we'll be able
to dynamically turn them on/off in SPDK makefiles.

Change-Id: I0ea639905360712361f34aeeb13827d5f48b4ac5
Reviewed-on: https://review.gerrithub.io/c/spdk/dpdk/+/460527 (spdk-19.05)
Reviewed-on: https://review.gerrithub.io/c/spdk/dpdk/+/471064 (spdk-19.08)
Reviewed-on: https://review.gerrithub.io/c/spdk/dpdk/+/477011 (spdk-19.11)
Signed-off-by: Darek Stojaczyk <dariusz.stojaczyk@intel.com>
Signed-off-by: Tomasz Zawadzki <tomasz.zawadzki@intel.com>
Reviewed-on: https://review.spdk.io/gerrit/c/spdk/dpdk/+/2531
Reviewed-by: Paul Luse <paul.e.luse@intel.com>
Reviewed-by: Jim Harris <james.r.harris@intel.com>
Reviewed-by: Ben Walker <benjamin.walker@intel.com>
Tested-by: Jim Harris <james.r.harris@intel.com>
---
 config/common_base  | 10 +++++-----
 config/common_linux |  2 +-
 2 files changed, 6 insertions(+), 6 deletions(-)

diff --git a/config/common_base b/config/common_base
index db3ce49b8..cc6fd70d5 100644
--- a/config/common_base
+++ b/config/common_base
@@ -143,7 +143,7 @@ CONFIG_RTE_LIBRTE_KVARGS=y
 #
 # Compile generic ethernet library
 #
-CONFIG_RTE_LIBRTE_ETHER=y
+#CONFIG_RTE_LIBRTE_ETHER=n
 CONFIG_RTE_LIBRTE_ETHDEV_DEBUG=n
 CONFIG_RTE_MAX_ETHPORTS=32
 CONFIG_RTE_MAX_QUEUES_PER_PORT=1024
@@ -858,13 +858,13 @@ CONFIG_RTE_LIBRTE_CFGFILE=n
 #
 # Compile librte_cmdline
 #
-CONFIG_RTE_LIBRTE_CMDLINE=y
+#CONFIG_RTE_LIBRTE_CMDLINE=n
 CONFIG_RTE_LIBRTE_CMDLINE_DEBUG=n
 
 #
 # Compile librte_hash
 #
-CONFIG_RTE_LIBRTE_HASH=n
+#CONFIG_RTE_LIBRTE_HASH=n
 CONFIG_RTE_LIBRTE_HASH_DEBUG=n
 
 #
@@ -964,7 +964,7 @@ CONFIG_RTE_LIBRTE_GSO=n
 #
 # Compile librte_meter
 #
-CONFIG_RTE_LIBRTE_METER=y
+#CONFIG_RTE_LIBRTE_METER=n
 
 #
 # Compile librte_classify
@@ -1027,7 +1027,7 @@ CONFIG_RTE_LIBRTE_PDUMP=n
 #
 # Compile vhost user library
 #
-CONFIG_RTE_LIBRTE_VHOST=n
+#CONFIG_RTE_LIBRTE_VHOST=n
 CONFIG_RTE_LIBRTE_VHOST_NUMA=n
 CONFIG_RTE_LIBRTE_VHOST_DEBUG=n
 
diff --git a/config/common_linux b/config/common_linux
index ba45de7bb..613c9f713 100644
--- a/config/common_linux
+++ b/config/common_linux
@@ -13,7 +13,7 @@ CONFIG_RTE_EAL_VFIO=y
 CONFIG_RTE_KNI_KMOD=n
 CONFIG_RTE_LIBRTE_KNI=n
 CONFIG_RTE_LIBRTE_PMD_KNI=n
-CONFIG_RTE_LIBRTE_VHOST=n
+#CONFIG_RTE_LIBRTE_VHOST=n
 CONFIG_RTE_LIBRTE_VHOST_NUMA=y
 CONFIG_RTE_LIBRTE_VHOST_POSTCOPY=n
 CONFIG_RTE_LIBRTE_PMD_VHOST=n
-- 
2.29.2

